cmds/cpud/main_freebsd.go

```diff
@@ -0,0 +1,131 @@
+	port        = flag.String("sp", "17010", "cpu default port")
+
+	debug     = flag.Bool("d", false, "enable debug prints")
+	runAsInit = flag.Bool("init", false, "run as init (Debug only; normal test is if we are pid 1")
+	// v allows debug printing.
+	// Do not call it directly, call verbose instead.
+	v       = func(string, ...interface{}) {}
+	remote  = flag.Bool("remote", false, "indicates we are the remote side of the cpu session")
+	network = flag.String("net", "tcp", "network to use")
+	port9p  = flag.String("port9p", "", "port9p # on remote machine for 9p mount")
+	klog    = flag.Bool("klog", false, "Log cpud messages in kernel log, not stdout")
+
+	// Some networks are not well behaved, and for them we implement registration.
+	registerAddr = flag.String("register", "", "address and port to register with after listen on cpu server port")
+	registerTO   = flag.Duration("registerTO", time.Duration(5*time.Second), "time.Duration for Dial address for registering")
+
+	// if we start up too quickly, mDNS won't work correctly.
+	// This sleep may be useful for other cases, so it is here,
+	// not specifically for mDNS uses.
+	sleepBeforeServing = flag.Duration("sleepBeforeServing", 0, "add a sleep before serving -- usually only needed if cpud runs as init with mDNS")
+
+	pid1 bool
+)
+
+func verbose(f string, a ...interface{}) {
+	if *remote {
+		v("CPUD(remote):"+f+"\r\n", a...)
+	} else {
+		v("CPUD:"+f, a...)
+	}
+}
+
+// There are three distinct cases to cover.
+//  1. running as init (indicated by pid == 1 OR -init=true switch
+//  2. running as server. pid != 1 AND -remote=true AND -init=false
+//  3. running as 'remote', i.e. the thing that starts a command for
+//     a client. Indicated by remote=true.
+//
+// case (3) overrides case 2 and 1.
+// This has evolved over the years, and, likely, the init and remote
+// switches ought to be renamed to 'role'. But so it goes.
+// The rules on arguments are very strict now. In the remote case,
+// os.Args[1] MUST be remote; no other invocation is accepted, because
+// the args to remote and the args to server are different.
+// This invocation requirement is known to the server package.
+func main() {
+	if len(os.Args) > 1 && (os.Args[1] == "-remote" || os.Args[1] == "-remote=true") {
+		*remote = true
+	}
+
+	if *remote {
+		// remote has far fewer args. Since they are specified by the client,
+		// we want to limit the set of args it can set.
+		flag.CommandLine = flag.NewFlagSet("cpud-remote", flag.ExitOnError)
+		debug = flag.Bool("d", false, "enable debug prints")
+		remote = flag.Bool("remote", false, "indicates we are the remote side of the cpu session")
+		port9p = flag.String("port9p", "", "port9p # on remote machine for 9p mount")
+
+		flag.Parse()
+		if *debug {
+			v = log.Printf
+			session.SetVerbose(verbose)
+		}
+		// If we are here, no matter what they may set, *remote must be true.
+		// sadly, cpud -d -remote=true -remote=false ... works.
+		*remote = true
```

Comment:
no sure if I get this, but for me line 69 defines remote value, maybe we should not parse it at all? I mean remove `remote = flag.Bool("remote", false, "indicates we are the remote side of the cpu session")`

Reply:
If we can ever run as init on freebsd, this will matter, so I'd rather leave it in. It follows the pattern of the linux side.